contracts/Registry.vy

```diff
@@ -91,6 +101,12 @@ pool_count: public(uint256)         # actual length of pool_list
 
 pool_data: HashMap[address, PoolArray]
 
+coin_count: public(uint256)  # total unique coins registered
+coins: HashMap[address, CoinInfo]
+get_coin: public(address[65536])  # unique list of registered coins
+coin_swap_indexes: HashMap[address, HashMap[address, uint256]]  # coina -> coinb -> pos in coin.swap_for
+coin_swap_register_count: HashMap[address, HashMap[address, uint256]]  # coina -> coinb -> times registered
```

Comment:
You could be using [`bitwise_xor`](https://vyper.readthedocs.io/en/stable/built-in-functions.html#bitwise_xor) here to avoid storing identical information twice - see the logic around `markets` for an example.

Reply:
Actually I think `coin_swap_register_count` might be achieving the same thing as `market_counts` already does?

Reply:
Right! Took some tinkering around but I refactored the logic to use the `market_counts` var. :+1: